src/ripple/app/main/Application.cpp

```diff
@@ -354,13 +354,31 @@ class ApplicationImp
 
     static
     std::size_t
-    numberOfThreads(Config const& config)
+    asioThreadCount(Config const& config)
     {
-    #if RIPPLE_SINGLE_IO_SERVICE_THREAD
+#if RIPPLE_SINGLE_IO_SERVICE_THREAD
         return 1;
-    #else
+#else
         return (config.NODE_SIZE >= 2) ? 2 : 1;
-    #endif
+#endif
+    }
+
+    // Calculate the number of threads for the job queue
+    // taking into account the configured node size for
+    // this instance.
+    static
+    unsigned int
+    jobThreadCount (Config const& config)
+    {
+        if (config.standalone())
+            return 1;
+
+        unsigned int const minThreads = 2;
+        unsigned int const maxThreads = 2 + (config.NODE_SIZE * 4);
```

Comment:
The 18 limit excludes the higher number of cores on newer server CPUs. eg. AMD Epic has 64 core count, Intel E7 40 core count.